pkg/image/registryclient/client.go

```diff
@@ -102,6 +104,7 @@ func (c *Context) Copy() *Context {
 		Limiter:           c.Limiter,
 
 		DisableDigestVerification: c.DisableDigestVerification,
+		AlternativeImageSources:   c.AlternativeImageSources,
```

Comment:
This needs to be an interface, not a string array.  By doing a string array, you are forcing the caller of the context to up front know what the alternatives for.  Instead, you should accept an interface like:

```
type AlternateImageSource interface {
  AlternateImageSource(registry *url.URL, repoName string, tag, digest string) ([]imagereference.Reference, error)
  ...
}
```

that provides you the list of alternates.  Then you have to calculate that at the time you hit the error, not in the `Repository()` call.

Reply:
updated - moved the alternative sources gathering here, to library-go, as a method of Context.  The Alternative Sources are now only a field in Context here, not in oc SecurityOptions. 
To get alternative sources (populate Context.AlternativeSources), caller has to provide a list of ImageContentSourcePolicies and an imageRef.

Reply:
unit tests exercising ^^ are currently in the oc PR

Reply:
I added the interface, ptal

Reply:
removed the interface in favor of adding all functions to library-go, as suggested

Reply:
added interface and moved all icsp fns to new package, as suggested